distributed/tests/test_worker.py

```diff
@@ -1366,21 +1366,19 @@ async def test_register_worker_callbacks_err(c, s, a, b):
 
 
 @gen_cluster(nthreads=[])
-async def test_local_directory(s):
-    with tmpfile() as fn:
-        with dask.config.set(temporary_directory=fn):
-            w = await Worker(s.address)
-            assert w.local_directory.startswith(fn)
-            assert "dask-worker-space" in w.local_directory
+async def test_local_directory(s, tmp_path):
+    with dask.config.set(temporary_directory=str(tmp_path)):
+        w = await Worker(s.address)
+        assert w.local_directory.startswith(str(tmp_path))
```

Comment:
Fun fact: This can be done a little bit nicer by `pathlib.Path(w.local_directory)` and then using the pathlib https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.is_relative_to if someone can sort out a release of the pathlib2 backport